docs/models.rst

```diff
@@ -76,14 +72,14 @@ function.  However the foreign key has to be separately declared with the
 
     class Person(db.Model):
         id = db.Column(db.Integer, primary_key=True)
-        name = db.Column(db.String(50))
-        addresses = db.relationship('Address', backref='person',
-                                    lazy='dynamic')
+        name = db.Column(db.String(50), nullable=False)
+        addresses = db.relationship('Address', backref='person', lazy=True)
 
     class Address(db.Model):
         id = db.Column(db.Integer, primary_key=True)
-        email = db.Column(db.String(50))
-        person_id = db.Column(db.Integer, db.ForeignKey('person.id'))
+        email = db.Column(db.String(120), nullable=False)
+        person_id = db.Column(db.Integer, db.ForeignKey('person.id'),
```

Comment:
0.8 is still supported :/ If it was unreliable there it's probably better to keep the type here for now and remove it once support for ancient SQLAlchemy Versions is dropped. FWIW, I think a Flask-SQLAlchemy==3.0 should switch to SQLAlchemy 1.0 at least, or even better 1.1.